pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -1017,6 +1021,10 @@ private CompletableFuture<Subscription> getDurableSubscription(String subscripti
         }
 
         Map<String, Long> properties = PersistentSubscription.getBaseCursorProperties(replicated);
+        if (readCompacted) {
+            properties = new TreeMap<>(properties);
+            properties.put(READ_COMPACTED_CURSOR_PROPERTIES, 1L);
```

Comment:
Why did you create a new map instead of modifying the `properties`? (the same question as it is in `readMoreEntries`)

```java
        if (readCompacted) {
            properties = new HashMap<>(properties);
            properties.put(READ_COMPACTED_CURSOR_PROPERTIES, 1L);
        }
```

Reply:
We need to merge customer properties but original map may be an immutable map.

Reply:
Yeah, we can create a new map and copy the elements from the immutable map.